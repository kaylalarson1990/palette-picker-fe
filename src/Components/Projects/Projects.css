.project-section {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.palette-box {
  align-content: center;
  display: flex;
  justify-content: space-around;
  width: 300px;
}

.box-color {
  border: 1px solid;
  height: 40px;
  width: 15%
}