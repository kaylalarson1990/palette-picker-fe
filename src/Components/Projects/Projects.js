import React, { Component } from "react";
import ColorSwatch from "../ColorSwatch/ColorSwatch";
import { connect } from "react-redux";

const Projects = props => {
    console.log('2', props)
    const savedPalettes = props.palettes.filter(
      palette => palette.project_id === props.project.id
    );
    const palettes = savedPalettes.map(palette => (
      <ColorSwatch data={palette} key={palette.id} projectID={props.project.id} />
    ));
    return (
      <section className="Project">
        <div className="project-header">
          <div>
            <h3 className="project-title">{props.project.project_name}</h3>
          </div>
          {/* <button
            onClick={() => props.fetchDeleteProject(props.project.id)}
          >
            <i class="fas fa-trash-alt trash-btn fa-2x" />
          </button> */}
        </div>
        <div className="project-palettes">{palettes}</div>
      </section>
    );
};

export const mapStateToProps = state => ({
  palettes: state.palettes
});

export default connect(mapStateToProps)(Projects);
