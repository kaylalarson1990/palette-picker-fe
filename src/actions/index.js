export const gatherProjects = projects => ({
  type: 'GATHER_PROJECTS',
  projects
})

export const gatherPalettes = palettes => ({
  type: 'GATHER_PALETTES',
  palettes
})

export const gatherProject = id => ({
  type: 'GATHER_PROJECT',
  id
})

export const gatherPalette = id => ({
  type: 'GATHER_PALETTE',
  id
})