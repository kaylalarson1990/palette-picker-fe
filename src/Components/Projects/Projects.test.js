import React from "react";
import { shallow } from "enzyme";
import { Projects, mapDispatchToProps, mapStateToProps } from "./Projects";
import { gatherPalettes, gatherProjects } from "../../actions/index";
import "../../../setupTests";

describe("Projects", () => {
  let wrapper, instance, props;
  beforeEach(() => {
    props = {
      palettes: [
        {
          palette_name: "Test Palette",
          c1: "Red",
          c2: "Blue",
          c3: "Green",
          c4: "Yellow",
          c5: "White",
          id: 1
        }
      ]
    };
    wrapper = shallow(<Projects {...props} />);
    instance = wrapper.instance();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should dispatch gatherPalettes when delete is clicked", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = gatherPalettes({
      id: 2
    });

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.gatherPalettes({
      id: 2
    });

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it("should dispatch gatherProjects when delete is clicked", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = gatherProjects({
      id: 2
    });

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.gatherProjects({
      id: 2
    });

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});

describe("mapStateToProps", () => {
  it("should return projects, palettes, and error from store", () => {
    const mockState = {
      palettes: [{ palette_name: "Spring Fling" }]
    };

    const expected = {
      palettes: [{ palette_name: "Spring Fling" }]
    };

    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe("mapDispatchToProps", () => {
  let mockDispatch;

  beforeEach(() => {
    mockDispatch = jest.fn();
  });

  it("should dispatch gatherProjects when handleSubmit is called", () => {
    const actionToDispatch = gatherProjects({
      projects: { project_name: "Test" }
    });

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.gatherProjects({
      projects: { project_name: "Test" }
    });

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it("should dispatch gatherPalettes when handleSubmit is called", () => {
    const actionToDispatch = gatherPalettes({
      palettes: { palette_name: "Test" }
    });

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.gatherPalettes({
      palettes: { palette_name: "Test" }
    });

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
