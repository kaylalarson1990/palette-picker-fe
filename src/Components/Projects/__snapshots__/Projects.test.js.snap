// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects should match the snapshot 1`] = `
<section
  className="project-section"
>
  <div
    className="project-info"
  >
    <h3 />
    <p
      className="delete-btn"
      onClick={[Function]}
    >
      ðŸ—‘
    </p>
  </div>
  <section>
    <Connect(Palette)
      c1="Red"
      c2="Blue"
      c3="Green"
      c4="Yellow"
      c5="White"
      id={1}
      key="1"
      palette_name="Test Palette"
    />
  </section>
</section>
`;
